### DISCUSSION TITLE
Dockershim

### DISCUSSION THREAD

**Discussion by adriangabura** [k3s-io/k3s#5848] [General]
Hi
What does it mean: "k3s will no longer support the --docker flag."?
Is curl -sfL https://get.k3s.io | sh -s - --docker still works or not?

Thanks!

**Comment by brandond** [comment]
https://kubernetes.io/blog/2022/02/17/dockershim-faq/

Kubernetes 1.24 no longer includes dockershim, so the --docker flag will not work with 1.24

**Reply by adriangabura** [reply]
Well, there's cri-dockerd which is developed by Mirantis, the new Docker Enterprise owners which they develop now outside Kubernetes project. I mean, it's a bit confusing for people just new to Kubernetes and are just setting up home labs for the first time. I thought Docker is not supported anymore, but it is, provided a correct CRI is provided.

**Reply by adriangabura** [reply]
Although I gotta say this whole story makes me chilly a bit. Because that makes the whole deal very funky and the future is very uncertain, I think. Anyway, all I care about really is Dockerfiles, because I put effort into learning how to deal with them and now I am concerned the knowledge will be obsolete.

**Reply by brandond** [reply]
Dockerfiles and docker images have nothing to do with the Docker runtime. As it says in the FAQ I linked:

>Are Docker and containers the same thing?
>Docker popularized the Linux containers pattern and has been instrumental in developing the underlying technology, however containers in Linux have existed for a long time. The container ecosystem has grown to be much broader than just Docker. Standards like OCI and CRI have helped many tools grow and thrive in our ecosystem, some replacing aspects of Docker while others enhance existing functionality.
>
> Will my existing container images still work?
> Yes, the images produced from docker build will work with all CRI implementations. All your existing images will still work exactly the same.

**Reply by MayanNetwork** [reply]
I am curious.  I have my K3S server running (on Centos Stream 8 VM).   Do I need to install Docker-ce?  I don't have anythin docker related, the fresh VM just has K3S (and a couple of packages it requested)

The reason I ask, is I am experimenting.  I only have access to K3S Source.tar.gz, so I assume I have to build it.  If I run make, it complains 

/.dapper ci
**FATA**[0000] exec: "docker": executable file not found in $PATH

**Reply by brandond** [reply]
You need docker to build k3s, but not to run it. Ironic, I know.
